### PR TITLE
client: add proxy option

### PR DESCRIPTION
**Pull request by ben-toogood**

### CODE REVIEW

**Comment by vtolstov** [comment]
why adding new arg to LookupRoute ? Why not create CallOption for proxy address? I think that it more idiomatic

**Comment by asim** [comment]
Set as call option as @vtolstov  mentioned. Basically set as the address in call opts.

File: client/options.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ type Options struct {
 	// Other options for implementations of the interface
 	// can be stored in a context
 	Context context.Context
+
+	// ProxyAddress to send requests via
+	ProxyAddress string
```
Comment (asim) [inline client/options.go]: Just cal Proxy and comment // Address to proxy all requests through

Comment (vtolstov) [reply client/options.go]: why not say Address and comment about that this is endpoint address or some proxy?

**Comment by ben-toogood** [comment]
@asim / @vtolstov I've actioned feedback.

**Comment by vtolstov** [comment]
LGTM, waiting @asim response

**Review by asim — APPROVED** [review]

**Comment by asim** [comment]
Looks good, merging. Happy to get rid of the env vars.

**Pull request merged**